### ISSUE TITLE
camera-bevpool for LSSTransform

### ISSUE THREAD

**Issue by sandeepnmenon** [NVIDIA-AI-IOT/Lidar_AI_Solution#144]
The cuda code given in `src/bevfusion/camera-bevpool.cu` is for the `SubclassCameraModule` mentioned in `qat/export-camera.py` which is for the `DepthLSSTransform` module.
I am trying to integrate the mask segmentation model `bevfusion-seg.pth` which uses `LSSTransform` and there is no depth.
I have rewritten many of the modules but not able to figure out how the `src/bevfusion/camera-bevpool.cu` should be changed to run the `LSSTransform` module.
I exported the module using
```python
class SubclassLSSTransformModule(nn.Module):
    def __init__(self, model):
        super(SubclassLSSTransformModule, self).__init__()
        self.model = model

    def forward(self, img):
        B, N, C, H, W = img.size()
        img = img.view(B * N, C, H, W)

        feat = self.model.encoders.camera.backbone(img)
        feat = self.model.encoders.camera.neck(feat)
        if not isinstance(feat, torch.Tensor):
            feat = feat[0]

        BN, C, H, W = map(int, feat.size())
        feat = feat.view(B, int(BN / B), C, H, W)

        def get_lss_feats(self, x):
            B, N, C, fH, fW = map(int, x.shape)
            x = x.view(B * N, C, fH, fW)

            x = self.depthnet(x)
            depth = x[:, : self.D].softmax(dim=1)
            x = depth.unsqueeze(1) * x[:, self.D : (self.D + self.C)].unsqueeze(2)
            x = x.view(B, N, self.C, self.D, fH, fW)
            x = x.permute(0, 1, 3, 4, 5, 2)

            return x
        
        return get_lss_feats(self.model.encoders.camera.vtransform, feat)

camera_model = SubclassLSSTransformModule(model)
camera_model.cuda().eval()
...
torch.onnx.export(
            camera_model,
            (img),
            camera_backbone_onnx,
            input_names=["img"],
            output_names=["camera_feature"],
            opset_version=16,
            do_constant_folding=True,
        )

```
`img`,  is same as used for the `SubclassCameraModule`

Should I return the depth from the `depthnet` same as in the `DepthLSSTransform` and use the same bevpool or write another one which does not use depth.

**Comment by WelY1** [comment]
Hi, is there any new progress?
